bugfix: Fixed the toolbar menu toggler and outside click

vikash-prem123 opened the pull request:
This PR fixed  handling the toolbar toggle and outside click on "Personal tools".

Fix  #5645  resolve issue with toolbar menu toggler

- refactor: update handleClickOutside in Toolbar component

-   Added a simple condition to not execute the "closeMenu" in handleClickOutside function when click on the "icon" because it will execute by toggleMenu function.

**Screen Recording**

[fix-toggle-menu.webm](https://github.com/plone/volto/assets/106796672/ae2c2c31-a836-48f5-b487-547a9ba6c4d3)

@vikash-prem123

mister-roboto commented:
@vikash-prem123 you need to sign the Plone Contributor Agreement to merge this pull request. 

Learn about the Plone Contributor Agreement: https://plone.org/foundation/contributors-agreement

stevepiercy commented:
@vikash-prem123 please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

vikash-prem123 commented:
I have carefully reviewed the [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html) documentation. I've addressed the feedback and corrected the identified mistakes in my pull request. Could you please review the changes and provide any further feedback? Thank you.

stevepiercy commented:
@vikash-prem123 you may have carefully read it, but you clearly did not follow at least two things in it, as indicated by the two failing workflows, specifically:

- [Sign and return the Plone Contributor Agreement](https://6.docs.plone.org/contributing/index.html#sign-and-return-the-plone-contributor-agreement)
- [Change log entry](https://6.docs.plone.org/contributing/index.html#change-log-entry)

Until those essential requirements are satisfied, no one will review this PR.

stevepiercy commented:
@vikash-prem123 we do not review pull requests that do not pass all tests. You should do that locally before creating a pull request. See [Code quality](https://6.docs.plone.org/volto/contributing/index.html#code-quality).

We also do not guide or mentor contributors.

vikash-prem123 commented:
@stevepiercy  plz review my PR now.

stevepiercy commented:
@vikash-prem123 what issue does this fix? You refer to this pull request, not an issue.

vikash-prem123 commented:
Fix #5645

stevepiercy commented on packages/volto/news/5645.bugfix:
```diff
@@ -0,0 +1 @@
+Added a simple condition to not execute the "closeMenu" in handleClickOutside function when click on the "icon" because it will execute by toggleMenu function. @vikash-prem123
```
News items usually should make sense to both end users and developers. Implementation details are obvious when a developer reads the diff.

```suggestion
The user personal tools menu now closes when clicked a second time or when clicking outside of the component. @vikash-prem123
```

stevepiercy reviewed (changes requested):
This also needs review by a core developer. In another pull request, @sneridagh referred to this https://github.com/plone/volto/blob/main/packages/volto/src/helpers/Utils/useDetectClickOutside.js. Maybe @ichim-david would know?

stevepiercy reviewed (commented):
News item is fine. Still needs review by a core developer. Thank you!

vikash-prem123 commented:
@stevepiercy  who will merge my PR?

ichim-david reviewed (changes requested):
@vikash-prem123 your fix isn't good enough, there are situations where you click on the SVG logos and target instead of being the SVG with the "icon" class you actually click on the path of the SVG so you still get the re-open

vikash-prem123 commented:
> @vikash-prem123 your fix isn't good enough, there are situations where you click on the SVG logos and target instead of being the SVG with the "icon" class you actually click on the path of the SVG so you still get the re-open

Ya I can see that thanks for your replay. I will fix it as soon as possible.

vikash-prem123 commented:
@ichim-david I have fixed that issue. please check it now.

stevepiercy reviewed (commented):
I approve the news item only.

ichim-david commented:
@vikash-prem123 seeing your suggested fix of modifying the user icon to add a class that you can hook on makes me see that you haven't understood the problem properly or taken my feedback into consideration as you should.
I could continue to give more things todo but then I end up fixing the issue while telling you how todo it bit by bit.

I will open a pull request with a fix and I welcome your eyes to observe what I did so that next time you do the proper testing and work so that any of the toggle buttons are fixed.

You should not rush to try and contribute to open source without having a good understanding of the product. I've worked with it for 2 years before I started to contribute and only after having real-world experience with things that require changing. 
Although everyone's path is different you need to have good experience with the product before you can make contributions and I suggest you take the time to learn it and use it before attempting to fix things.

vikash-prem123 commented:
@ichim-david  Thank you for your feedback. I understand the importance of gaining more experience with the product before contributing to open-source projects. I appreciate your insights and look forward to reviewing your pull request to learn from your changes. I'm open to collaboration and appreciate your guidance.
